package/src/components/Attachment/__tests__/Giphy.test.js

```diff
@@ -0,0 +1,290 @@
+        height: '200',
+        url: 'https://media1.giphy.com/media/test/fixed_height.gif',
+        width: '375',
+      },
+      original: {
+        height: '256',
+        url: 'https://media1.giphy.com/media/test/original.gif',
+        width: '480',
+      },
+    };
+    const { getByTestId: getByTestIdFixedHeight } = render(
+      getAttachmentComponent({ attachment, giphyVersion: 'fixed_height' }),
+    );
+    const { getByTestId: getByTestIdOriginal } = render(
+      getAttachmentComponent({ attachment, giphyVersion: 'original' }),
+    );
+    await waitFor(() => {
+      const checkImageProps = (imageProps, specificSizedGiphyData) => {
+        let imageStyle = imageProps.style;
+        if (Array.isArray(imageStyle)) {
+          imageStyle = Object.assign({}, ...imageStyle);
+        }
+        expect(imageStyle.height).toBe(parseFloat(specificSizedGiphyData.height));
+        expect(imageStyle.width).toBe(parseFloat(specificSizedGiphyData.width));
+        expect(imageProps.source.uri).toBe(specificSizedGiphyData.url);
+      };
+      checkImageProps(
+        getByTestIdFixedHeight('giphy-attachment-image').props,
+        attachment.giphy.fixed_height,
+      );
+      checkImageProps(
+        getByTestIdOriginal('giphy-attachment-image').props,
+        attachment.giphy.original,
+      );
+    });
+  });
+
+  it('show render giphy action UI and all the 3 action buttons', async () => {
+    const attachment = generateGiphyAttachment();
+    attachment.actions = [
+      { name: 'image_action', text: 'Send', value: 'send' },
+      { name: 'image_action', text: 'Shuffle', value: 'shuffle' },
+      {
+        name: 'image_action',
+        text: 'Cancel',
+        value: 'cancel',
+      },
+    ];
+    const { getByTestId } = render(
+      getAttachmentComponent({ attachment, giphyVersion: 'fixed_height' }),
+    );
+
+    await waitFor(() => {
+      expect(getByTestId('giphy-action-attachment')).toBeTruthy();
+      expect(getByTestId('cancel-action-button')).toBeTruthy();
+      expect(getByTestId('shuffle-action-button')).toBeTruthy();
+      expect(getByTestId('send-action-button')).toBeTruthy();
+    });
+  });
+
+  it('should trigger the cancel giphy action', async () => {
+    const attachment = generateGiphyAttachment();
+    const handleAction = jest.fn();
+    attachment.actions = [
+      { name: 'image_action', text: 'Send', value: 'send' },
+      { name: 'image_action', text: 'Shuffle', value: 'shuffle' },
+      {
+        name: 'image_action',
+        text: 'Cancel',
+        value: 'cancel',
+      },
+    ];
+    const { getByTestId } = render(
+      getAttachmentComponent({
+        attachment,
+        giphyVersion: 'fixed_height',
+        handleAction,
+      }),
+    );
+
+    await waitFor(() => getByTestId(`${attachment.actions[2].value}-action-button`));
+
+    expect(getByTestId('giphy-action-attachment')).toContainElement(
+      getByTestId(`${attachment.actions[2].value}-action-button`),
```

Comment:
I think we should simply check by text of the button - "Cancel", "Shuffle", "Send"!!